### PR TITLE
Fixes module combo box and button state

### PR DESCRIPTION
**Pull request by joaoandremartins**
Correctly disables the modules combo box and module settings button if both app.yaml and Dockerfile overrides are turned on, and enables them if otherwise.

### CODE REVIEW

**Comment by joaoandremartins** [comment]
Right now, only the module combo box gets disabled on resetForm and then it isn't possible to change its state without reopening the window.

**Comment by etanshaul** [comment]
does this address a certain issue #?

**Comment by joaoandremartins** [comment]
I never got to creating an issue for this bug.

**Comment by etanshaul** [comment]
ok, no worries

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/flexible/AppEngineFlexibleDeploymentEditor.java [hunk 1]
```diff
@@ -158,6 +158,9 @@ protected void textChanged(DocumentEvent event) {
     appYamlOverrideCheckBox.addActionListener(event -> {
       boolean isAppYamlOverrideSelected = ((JCheckBox) event.getSource()).isSelected();
       appYamlTextField.setVisible(isAppYamlOverrideSelected);
+
+      setModuleComboBoxAndButtonEnabled(
```
Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/flexible/AppEngineFlexibleDeploymentEditor.java]: setModuleControls would be good

Comment (joaoandremartins) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/flexible/AppEngineFlexibleDeploymentEditor.java]: Why do you suggest?
The method only sets the module bits to enabled/disabled, and I was basing on other methods like setEnabled().
I can replace Enabled by State?

**Comment by etanshaul** [comment]
Its a more general issue, but its a little confusing that we don't disable the module controls when only checking "use a different app.yaml", but not Dockerfile. I say this because with the UI its not clear that the Dockerfile is also coming from the module. Probably not the right place to fix this though.

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/flexible/AppEngineFlexibleDeploymentEditor.java (hunk 1 above)
Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/flexible/AppEngineFlexibleDeploymentEditor.java]: yea I guess you're right. It felt a little weird with it being long. But I have no problem with setModuleControlsEnabled/State

**Comment by joaoandremartins** [comment]
Not sure if I know what you mean.
We also change the module elements state when "use a different dockerfile" is checked.

**Comment by etanshaul** [comment]
it appears you only change the state if _both_ are selected. I was referrring to the case where only "use another yaml" is selected.

**Comment by etanshaul** [comment]
my point being is that from the UI it only clear that module settings applies to app.yaml, not to Dockerfile. but like I said, not the place to fix this.

**Comment by joaoandremartins** [comment]
Aah yes. That's on purpose, because the Dockerfile is still being gotten from the module.
Though you're right, that relationship isn't downright visible.

**Pull request merged**